src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,62 @@
+package duke.task;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Represents a deadline item that inherits from Task.
+ */
+public class Deadline extends Task {
+
+    /** Date of when the deadline is due. */
+    protected Date date;
+    /** Boolean that represents if the date contains a time. */
+    protected boolean isTime;
+
+    /**
+     * Creates a deadline with a default not done.
+     * @param description Description of deadline.
+     * @param date Date of deadline.
+     * @param isTime Indicates if date has a time.
+     */
+    public Deadline(String description, Date date, boolean isTime) {
+        super(description);
+        this.date = date;
+        this.isTime = isTime;
+    }
+
+    /**
+     * Creates a deadline.
+     * @param description Description of deadline.
+     * @param date Date of deadline.
+     * @param isTime Indicates if date has a time.
+     * @param isDone Indicates if deadline is done.
+     */
+    public Deadline(String description, Date date, boolean isTime, boolean isDone) {
+        super(description, isDone);
+        this.date = date;
+        this.isTime = isTime;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return Deadline as a String.
+     */
+    @Override
+    public String toString() {
+        String s = isTime ? "MMM dd yyyy hh:mma" : "MMM dd yyyy";
```

Comment:
I like the use of the scratch variable `s` since the variable is used for temporary storage only. 👍